### PR TITLE
ITE: soc: it8xxx2: correct the offset of eflash signature

### PR DESCRIPTION
**Pull request by Dino-Li**
This change corrects the offset to 80h and also add description
of flash signature.

Signed-off-by: Dino Li <Dino.Li@ite.com.tw>

### CODE REVIEW

File: soc/riscv/riscv-ite/common/vector.S [hunk 1]
```diff
@@ -39,22 +39,35 @@ SECTION_FUNC(vectors, __start)
...
+.balign 16
+.global eflash_sig
+eflash_sig:
+.byte 0xA5, 0xA5, 0xA5, 0xA5, 0xA5, 0xA5
+#ifdef CONFIG_ESPI
+.byte 0xA4 /* enable eSPI */
+#else
+.byte 0xA5 /* enable LPC */
+#endif
+.byte 0xB4 /* flag of signature */
```
Comment (brockus-zephyr) [inline soc/riscv/riscv-ite/common/vector.S:72]: Is there a CONFIG_ that exists or could exist to make the flags change based on the internal clock generator being used or not?

Comment (Dino-Li) [reply soc/riscv/riscv-ite/common/vector.S]: I created a configuration option to select external/internal clock source.

**Review by brockus-zephyr — APPROVED** [review]

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc [hunk 2]
```diff
@@ -11,3 +11,12 @@ config SOC_IT8XXX2
 	select ATOMIC_OPERATIONS_C
 
 endchoice
+
+config SOC_IT8XXX2_EXT_32K
+	bool "Use external 32.768 kHz clock source"
+	default n
```
Comment (keith-zephyr) [inline soc/riscv/riscv-ite/it8xxx2/Kconfig.soc:17]: bool symbols implicitly default to n.  Zephyr style recommends skipping redundant defaults.
https://docs.zephyrproject.org/2.4.0/guides/kconfig/tips.html#redundant-defaults

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc (hunk 2 above)
Comment (Dino-Li) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.soc]: Done.

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc [hunk 3]
```diff
@@ -11,3 +11,11 @@ config SOC_IT8XXX2
...
 
 endchoice
+
+config SOC_IT8XXX2_EXT_32K
+	bool "Use external 32.768 kHz clock source"
+	help
+	  Use an external 32.768 kHz clock source for PLL reference
+	  clock. A 32.768 kHz oscillator is required if set y.
+
+	  Default n to use the +/-2.3% internal clock generator.
```
Comment (jackrosenthal) [inline soc/riscv/riscv-ite/it8xxx2/Kconfig.soc:21]: Seems like this should be a choice option instead of bool

i.e., one for internal, one for external

Comment (Dino-Li) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.soc]: Done.

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc [hunk 4]
```diff
@@ -11,3 +11,15 @@ config SOC_IT8XXX2
 	select ATOMIC_OPERATIONS_C
 
 endchoice
+
+choice
+	prompt "Select clock source for PLL reference clock"
+	default SOC_IT8XXX2_INT_32K
```
Comment (jackrosenthal) [inline soc/riscv/riscv-ite/it8xxx2/Kconfig.soc:17]: "default" can be dropped here. The default is always the first for choice symbols

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc [hunk 5]
```diff
@@ -11,3 +11,15 @@ config SOC_IT8XXX2
 	select ATOMIC_OPERATIONS_C
 
 endchoice
+
+choice
+	prompt "Select clock source for PLL reference clock"
```
Comment (jackrosenthal) [inline soc/riscv/riscv-ite/it8xxx2/Kconfig.soc:16]: "Clock source for PLL reference clock"

(drop "Select")

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc (hunk 4 above)
Comment (Dino-Li) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.soc]: Done

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.soc (hunk 5 above)
Comment (Dino-Li) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.soc]: Done

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**